File: my_py_pkg/my_py_pkg/add_two_ints_client.py
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


class AddTwoIntsClient(Node):
    def __init__(self):
        super().__init__('add_two_ints_client')

        self.call_add_two_ints_server(5, 6)
        self.get_logger().info('add_two_ints_client Started...')

    def call_add_two_ints_server(self, a, b):
        client_ = self.create_client(AddTwoInts, 'add_two_ints')
        while not client_.wait_for_service(1.0):
            self.get_logger().warning('Waiting for service...')

        req = AddTwoInts.Request()
        req.a = a
        req.b = b

        future = client_.call_async(req)
        future.add_done_callback(lambda future: self.callback_future_add_two_ints(future, req))

    def callback_future_add_two_ints(self, future, req):
        try:
            res = future.result()
            self.get_logger().info(f"{req.a} + {req.b} = {res.sum}")
        except Exception as e:
            self.get_logger().error(f"add_two_ints_client service call failed: %r", (e,))


def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsClient()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: my_cpp_pkg/src/number_system/add_two_ints_client.cpp
#include <rclcpp/rclcpp.hpp>
#include <example_interfaces/srv/add_two_ints.hpp>

using namespace std::chrono_literals;
using example_interfaces::srv::AddTwoInts;

class AddTwoIntsClient : public rclcpp::Node {
   private:
    rclcpp::Client<AddTwoInts>::SharedPtr client_;
    std::thread thread1_;

    void callAddTwoIntsServer(int a, int b) {
        while(!client_->wait_for_service(1s))
            RCLCPP_WARN(this->get_logger(), "Waiting for 'add_two_ints' server...");

        auto req = std::make_shared<AddTwoInts::Request>();
        req->a = a;
        req->b = b;

        auto future = client_->async_send_request(req);
        
        if (future.wait_for(5s) == std::future_status::ready) {
            try{
                auto res = future.get();
                RCLCPP_INFO(this->get_logger(), "%ld + %ld = %ld", req->a, req->b, res->sum);
            } catch (const std::exception &err) {
                RCLCPP_ERROR(this->get_logger(), "Service call 'add_two_ints' failed, %s", err.what());
            }
        } else {
            RCLCPP_ERROR(this->get_logger(), "Service call 'add_two_ints' timed out.");
        }
    }

   public:
    AddTwoIntsClient() : Node("add_two_ints_client")
    {
        client_ = this->create_client<AddTwoInts>("add_two_ints");
        thread1_ = std::thread(std::bind(&AddTwoIntsClient::callAddTwoIntsServer, this, 1,  4));
    }
};

int main(int argc, char** argv) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<AddTwoIntsClient>());
    rclcpp::shutdown();
    return 0;
}